applications/luci-app-shadowsocks-libev/htdocs/luci-static/resources/shadowsocks-libev.js

```diff
@@ -0,0 +1,181 @@
+	'camellia-128-cfb',
+	'camellia-192-cfb',
+	'camellia-256-cfb',
+	'salsa20',
+	'chacha20',
+	'chacha20-ietf',
+];
+
+return L.Class.extend({
+	values_actions: function(o) {
+		o.value('bypass');
+		o.value('forward');
+		if (o.option !== 'dst_default') {
+			o.value('checkdst');
+		}
+	},
+	values_redir: function(o, xmode) {
+		uci.sections('shadowsocks-libev', 'ss_redir', function(sdata) {
+			var disabled = this.ucival_to_bool(sdata['disabled']),
+				sname = sdata['.name'],
+				mode = sdata['mode'] || 'tcp_only';
+			if (!disabled && mode.indexOf(xmode) !== -1) {
+				o.value(sname, sname + ' - ' + mode);
+			}
+		});
+		o.value('', '<unset>');
+		o.default = '';
+	},
+	options_server: function(s, opts) {
+		var o, optfunc,
+			tab = opts && opts.tab || null;
+
+		if (!tab) {
+			optfunc = function(/* ... */) {
+				var o = s.option.apply(s, arguments);
+				o.editable = true;
+				return o;
+			};
+		} else {
+			optfunc = function(/* ... */) {
+				var o = s.taboption.apply(s, [tab].concat(arguments));
+				o.editable = true;
+				return o;
+			};
+		}
+
+		o = optfunc(form.Value, 'server', _('Server'));
+		o.datatype = 'host';
+		o.size = 16;
+
+		o = optfunc(form.Value, 'server_port', _('Server port'));
+		o.datatype = 'port';
+		o.size = 5;
+
+		o = optfunc(form.ListValue, 'method', _('Method'));
+		methods.forEach(function(m) {
+			o.value(m);
+		});
+
+		o = optfunc(form.Value, 'password', _('Password'));
+		o.password = true;
+		o.size = 12;
+
+		o = optfunc(form.Value, 'key', _('Key (base64)'));
+		o.datatype = 'base64';
+		o.password = true;
+		o.size = 12;
+		o.modalonly = true;;
+
+		optfunc(form.Value, 'plugin', _('Plugin')).modalonly = true;
+
+		optfunc(form.Value, 'plugin_opts', _('Plugin Options')).modalonly = true;
+	},
+	ucival_to_bool: function(val) {
+		return val === 'true' || val === '1' || val === 'yes' || val === 'on';
+	},
+	cfgvalue_overview: function(sdata) {
+		var stype = sdata['.type'],
+			lines = [];
+
+		if (stype === 'ss_server') {
+			this.cfgvalue_overview_(sdata, lines, names_options_server);
+			this.cfgvalue_overview_(sdata, lines, names_options_common);
+			this.cfgvalue_overview_(sdata, lines, ['bind_address']);
+		} else if (stype === 'ss_local' || stype === 'ss_redir' || stype === 'ss_tunnel') {
+			this.cfgvalue_overview_(sdata, lines, names_options_client);
+			if (stype === 'ss_tunnel') {
+				this.cfgvalue_overview_(sdata, lines, ['tunnel_address']);
+			}
+			this.cfgvalue_overview_(sdata, lines, names_options_common);
+		} else {
+			return [];
+		}
+
+		return lines;
+	},
+	cfgvalue_overview_: function(sdata, lines, names) {
+		names.forEach(function(n) {
+			var v = sdata[n];
+			if (v) {
+				if (n === 'key' || n === 'password') {
+					v = _('<hidden>');
+				}
+				var fv = E('var', [v]);
+				if (sdata['.type'] !== 'ss_server' && n === 'server') {
+					fv = E('a', {
+						class: 'label',
+						href: L.url('admin/services/shadowsocks-libev/servers', v) // FIXME open edit dialog
```

Comment:
In server-side, this link will open config edit page. But in client-side, it will not open config edit dialog.

Reply:
Fixed this using location.hash `#edit=section_id`.
https://github.com/openwrt/luci/blob/f2dfe5f53040dbeb6b4d42664d2eb043d2ba1ced/applications/luci-app-shadowsocks-libev/htdocs/luci-static/resources/view/shadowsocks-libev/servers.js#L27-L36
I use addFooter because at this time the edit button is present in document.